lib/bolt/task/puppet_server.rb

```diff
@@ -11,19 +11,17 @@ def initialize(task_data, file_cache)
         super(task_data)
       end
 
-      # TODO: currently this both converts filename -> name and fetches files to generate the path
-      # In the future we should not generate paths until after the implementation is selected to avoid
-      # unecessary fetching of files
+      # puppetserver file entries have 'filename' rather then 'name'
       def update_file_data(task_data)
-        if task_data['files']
-          task_data['files'] = task_data['files'].map do |f|
-            { 'name' => f['filename'],
-              'path' => @file_cache.update_file(f) }
-          end
-        end
-
+        task_data['files'] = task_data['files'].map { |f| f['name'] = f['filename'] }
         task_data
       end
+
+      # Compute local path and download files from puppetserver as needed
+      def file_path(file_name)
+        file = file_map[file_name]
```

Comment:
Do we need to look up the file if `file['path']` is already set? Could use a `begin` block to wrap everything.

Reply:
We don't call update_file on the next line if path is set. I'm not sure what you're suggesting, this cuts out a line of code:
```
file_map[file_name]['path'] ||= @file_cache.update_file(file)
```